Firestore/Source/Core/FSTFirestoreClient.m

```diff
@@ -249,6 +258,59 @@ - (void)removeListener:(FSTQueryListener *)listener {
   }];
 }
 
+- (void)getDocumentFromLocalCache:(FIRDocumentReference *)doc
+                       completion:(void (^)(FIRDocumentSnapshot *_Nullable document,
+                                            NSError *_Nullable error))completion {
+  [self.workerDispatchQueue dispatchAsync:^{
+    FSTMaybeDocument *maybeDoc = [self.localStore readDocument:doc.key];
+    if (maybeDoc) {
+      completion([FIRDocumentSnapshot snapshotWithFirestore:doc.firestore
+                                                documentKey:doc.key
+                                                   document:(FSTDocument *)maybeDoc
+                                                  fromCache:YES],
+                 nil);
+    } else {
+      completion(
+          nil, [NSError errorWithDomain:FIRFirestoreErrorDomain
+                                   code:FIRFirestoreErrorCodeUnavailable
+                               userInfo:@{
+                                 NSLocalizedDescriptionKey : @"Failed to get document from server.",
```

Comment:
More error text to explain that this is a user-requested error condition (as above).

Also in this case, this is a cache-only request so the server has nothing to do with it, right?

Reply:
Re cache vs server: Oops.

Updated text:

"Failed to get document from cache. (However, this document may exist on the server. Run again without setting FIRCache in the GetOptions to attempt to retrieve the document from the server.)"